src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -875,233 +873,192 @@ public void setFlags(final int value) {
         setIndexingBin(null);
     }
 
-    /**
-     * the read is paired in sequencing, no matter whether it is mapped in a pair.
-     */
-    public boolean getReadPairedFlag() {
-        return (mFlags & SAMFlag.READ_PAIRED.flag) != 0;
-    }
+    ////////////////////////////////////////////////////////////////////////////
+    // Flag accessor and mutators
+    ////////////////////////////////////////////////////////////////////////////
 
-    private void requireReadPaired() {
-        if (!getReadPairedFlag()) {
-            throw new IllegalStateException("Inappropriate call if not paired read");
-        }
-    }
+    /** Returns true if the "read paired in sequencing" flag is true. */
+    public boolean isPaired() { return getFlag(SAMFlag.READ_PAIRED); }
 
-    /**
-     * the read is mapped in a proper pair (depends on the protocol, normally inferred during alignment).
-     */
-    public boolean getProperPairFlag() {
-        requireReadPaired();
-        return getProperPairFlagUnchecked();
-    }
+    /** Sets the "read paired in sequencing" flag to the given value. */
+    public void setPaired(boolean paired) { setFlag(SAMFlag.READ_PAIRED, paired); }
 
-    private boolean getProperPairFlagUnchecked() {
-        return (mFlags & SAMFlag.PROPER_PAIR.flag) != 0;
-    }
+    /** @deprecated since May 2017; use {@link #isPaired()} instead. */
+    @Deprecated public boolean getReadPairedFlag() { return isPaired(); }
+    /** @deprecated since May 2017; use {@link #setPaired(boolean)} instead. */
+    @Deprecated public void setReadPairedFlag(final boolean flag) { setPaired(flag); }
 
-    /**
-     * the query sequence itself is unmapped.
-     */
-    public boolean getReadUnmappedFlag() {
-        return (mFlags & SAMFlag.READ_UNMAPPED.flag) != 0;
-    }
+    private void requireReadPaired() { if (!isPaired()) throw new IllegalStateException("Inappropriate call if not paired read"); }
 
-    /**
-     * the mate is unmapped.
-     */
-    public boolean getMateUnmappedFlag() {
-        requireReadPaired();
-        return getMateUnmappedFlagUnchecked();
-    }
 
-    private boolean getMateUnmappedFlagUnchecked() {
-        return (mFlags & SAMFlag.MATE_UNMAPPED.flag) != 0;
-    }
+    /** Returns true if both reads in the read pair are aligned as expected. */
+    public boolean isProperlyPaired() { requireReadPaired(); return getFlag(SAMFlag.PROPER_PAIR); }
 
-    /**
-     * strand of the query (false for forward; true for reverse strand).
-     */
-    public boolean getReadNegativeStrandFlag() {
-        return (mFlags & SAMFlag.READ_REVERSE_STRAND.flag) != 0;
-    }
+    /** Sets whether both reads in the read pair are aligned as expected. */
+    public void setProperlyPaired(final boolean paired) { setFlag(SAMFlag.PROPER_PAIR, paired); }
```

Comment:
I think that ship has sailed, in general - it would be a breaking change to add those kinds of checks into the setter, and I don't want to be the one who does that ;)  More specifically, I think things are the way they are so that records can be cleaned up without having to worry about ordering.  E.g. if you want to make a record "unpaired", you don't have to worry about calling `setProperlyPaired(false)` before `setPaired(false)` etc.